File: src/main/java/com/dafttech/terra/game/world/entities/EntityTorchArrow.java
package com.dafttech.terra.game.world.entities;

import com.dafttech.terra.engine.Vector2;
import com.dafttech.terra.game.world.World;
import com.dafttech.terra.game.world.tiles.Tile;
import com.dafttech.terra.game.world.tiles.TileTorch;

public class EntityTorchArrow extends EntityFlamingArrow {

    public EntityTorchArrow(Vector2 pos, World world) {
        super(pos, world);
    }

    @Override
    public void placeBlockOnHit(int x, int y) {
        Tile spreadTile = worldObj.getTile(x, y);
        if (spreadTile.isReplacable()) {
            worldObj.setTile(x, y, new TileTorch(), true);
        }
    }
}

File: terra/src/main/scala/com/dafttech/terra/game/world/entities/EntityTorchArrow.scala
package com.dafttech.terra.game.world.entities

import com.dafttech.terra.engine.TilePosition
import com.dafttech.terra.engine.vector.Vector2d
import com.dafttech.terra.game.world.tiles.TileTorch
import com.dafttech.terra.game.world.{GameWorld, TileInstance}

class EntityTorchArrow(pos: Vector2d)(implicit world: GameWorld) extends EntityFlamingArrow(pos) {
  override def placeBlockOnHit(tilePosition: TilePosition): Unit = {
    val spreadTile = tilePosition.getTile
    if (spreadTile.tile.isReplacable) world.setTile(tilePosition.pos, TileInstance(new TileTorch()), notify = true)
  }
}